### PR TITLE
Add `--upstream-branch` option

### PR DESCRIPTION
**Pull request by andre-rosa**

### CODE REVIEW

File: superflore/generators/bitbake/run.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ def main():
         selected_targets = get_distros_by_status('active')
     repo_org = 'allenh1'
     repo_name = 'meta-ros'
+    branch_name = args.upstream_branch or ''
```
Comment (allenh1) [inline superflore/generators/bitbake/run.py]: I don't think you need the `or ''` at the end of this line, do you?

File: superflore/generators/ebuild/run.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ def main():
         selected_targets = get_distros_by_status('active')
     repo_org = 'ros'
     repo_name = 'ros-overlay'
+    branch_name = args.upstream_branch or ''
```
Comment (allenh1) [inline superflore/generators/ebuild/run.py]: Or here

Comment (andre-rosa) [reply superflore/generators/ebuild/run.py]: Done

File: superflore/generators/bitbake/run.py (hunk 1 above)
Comment (andre-rosa) [reply superflore/generators/bitbake/run.py]: Removed, thanks!

File: superflore/generators/ebuild/overlay_instance.py [hunk 3]
```diff
@@ -23,10 +23,12 @@
 
 
 class RosOverlay(object):
-    def __init__(self, repo_dir, do_clone, org='ros', repo='ros-overlay'):
+    def __init__(
+            self, repo_dir, do_clone, org='ros', repo='ros-overlay',
```
Comment (allenh1) [inline superflore/generators/ebuild/overlay_instance.py]: nit: I think you can reduce the indent of this line to keep the `from_branch=''` on the same line -- I would prefer if you could

**Comment by allenh1** [comment]
Coverage report for this branch:

```
Name                                               Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------
superflore/CacheManager.py                            20      0   100%
superflore/PackageMetadata.py                         22      0   100%
superflore/TempfileManager.py                         34      6    82%   48-56
superflore/__init__.py                                10      4    60%   5-8
superflore/docker.py                                  78     22    72%   42, 56-69, 75, 93-95, 102, 107-108, 119, 121-127
superflore/exceptions.py                              18      2    89%   33, 38
superflore/generate_installers.py                     65      2    97%   46-47
superflore/generators/__init__.py                      0      0   100%
superflore/generators/bitbake/__init__.py              3      1    67%   3
superflore/generators/bitbake/gen_packages.py        102     79    23%   40-122, 129-168, 175-183, 189
superflore/generators/bitbake/ros_meta.py             32     23    28%   26-30, 33-38, 41-54, 57-58
superflore/generators/bitbake/run.py                 123    100    19%   41-201
superflore/generators/bitbake/yocto_recipe.py        118     99    16%   44-67, 70-74, 77-89, 92-96, 99-101, 104-105, 113-116, 125-184
superflore/generators/ebuild/__init__.py               3      1    67%   3
superflore/generators/ebuild/ebuild.py               185     19    90%   96-99, 126-131, 146, 176-180, 186, 193, 206-210
superflore/generators/ebuild/gen_packages.py         140    113    19%   46-119, 125-137, 143-188, 193-210, 213, 216
superflore/generators/ebuild/metadata_xml.py          33      0   100%
superflore/generators/ebuild/overlay_instance.py      45     33    27%   29-34, 37-47, 52-75, 78-79
superflore/generators/ebuild/run.py                  119     97    18%   40-188
superflore/parser.py                                  15      0   100%
superflore/repo_instance.py                           70     51    27%   31-52, 59-68, 71-79, 85-86, 92, 98-99, 105, 108-123, 126
superflore/rosdep_support.py                          37      3    92%   44, 88-89
superflore/test_integration/__init__.py                0      0   100%
superflore/test_integration/gentoo/__init__.py         3      1    67%   3
superflore/test_integration/gentoo/build_base.py      28     19    32%   26-28, 33, 38-54
superflore/test_integration/gentoo/main.py            32     26    19%   24-80
superflore/utils.py                                  191     34    82%   57-60, 64-78, 82-93, 108, 217-218, 222, 231, 235, 237, 239, 241, 243, 245, 247, 249
--------------------------------------------------------------------------------
TOTAL                                               1526    735    52%
```

Still looking pretty good overall! @tfoote I don't think this affects it much, so I'm going to go ahead and approve this one.

I'll leave it to you to merge it.

File: superflore/generators/ebuild/overlay_instance.py (hunk 3 above)
Comment (andre-rosa) [reply superflore/generators/ebuild/overlay_instance.py]: Thanks! I applied the same format as regenerate_manifests below.

**Pull request merged**